agent/config/runtime_test.go

```diff
@@ -6876,6 +6891,7 @@ func TestSanitize(t *testing.T) {
 		"AEInterval": "0s",
 		"AdvertiseAddrLAN": "",
 		"AdvertiseAddrWAN": "",
+		"AdvertiseReconnectTimeout": "0s",
```

Comment:
I feel like there's a missing config parsing round trip test update for this flag (the one with lots of random strings and numbers in it).

Reply:
I think that is TestFullConfig but it doesn't blow up because at some point we had to disable the nonZero field checking. There is a comment about it. I updated that test anyways but had to have the hcl/json contain `advertise_reconnect_timeout = "0s"` because the config also set server mode to true. So it isn't really validating much extra unfortunately.